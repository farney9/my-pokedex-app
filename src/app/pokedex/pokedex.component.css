.container {
    margin: auto;
    padding: 20px 10px 20px 10px;
    max-width: 1200px;
    min-height: 100%;
    display: flex;
    flex-direction: column;
}

.header {
    width: 100%;
    background-color: var(--main-color);
    height: 100px;
    border-radius: 10px 10px 0 0;
    display: flex;
    justify-content: center;
    align-items: center;
}

.header {
    width: 100%;
    background-color: var(--main-color);
    height: 100px;
    border-radius: 10px 10px 0 0;
    display: flex;
    justify-content: center;
    align-items: center;
}

.header__logo{
    width: 90px;
}

.header__title{
    color: #fff;
    font-size: 1.8em;
}

.description {
    display: flex;
    flex-direction: column;
    height: 200px;
}

.description__title {
    background-color: var(--light-color);
    height: 60px;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 8px;
    font-size: 2.2em;
}

