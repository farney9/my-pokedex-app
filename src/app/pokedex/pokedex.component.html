<div class="container">
    <header class="header">
        <img src="../../assets/pokeball.png" alt="logo" class="header__logo">
        <h1 class="header__title">POKÉDEX</h1>

    </header>
    <main class="main">
        <div *ngIf="!charizard; else elseBlock">
            <h2>Loading ...</h2>
        </div>
        <aside class="description">
            <ng-template #elseBlock>

                <h2 class="description__title">
                    <span class="description__title__icon">
                        <i class="fa-solid fa-fire-flame-curved fire" id="icon"></i>
                    </span>
                    {{charizard?.name | titlecase}}
                </h2>

            </ng-template>


        </aside>
        <article *ngIf="charizard" class="information">
            <div class="description__container">
                <img class="description__image hvr-grow" id="image" [src]="charizard?.sprites?.other?.home?.front_default"
                    [alt]="charizard?.name">
            </div>
            <div class="details">
                <div class="information__group">
                    <span class="information__group__title">N°</span>
                    <span class="information__group__description" id="number">00{{charizard.id}}</span>
                </div>
                <div class="information__group">
                    <span class="information__group__title">LEVEL</span>
                    <span class="information__group__description" id="number">{{charizard.base_experience}}</span>
                </div>
                <div class="information__group">
                    <span class="information__group__title">TYPE</span>
                    <span class="information__group__description" id="number">{{charizard.types[0].type.name}}</span>
                </div>
                <div class="information__group">
                    <span class="information__group__title">ABILITY</span>
                    <span class="information__group__description" id="number">{{charizard.abilities[0].ability.name}}</span>
                </div>
                <div class="information__group">
                    <span class="information__group__title">HEIGHT</span>
                    <span class="information__group__description" id="number">{{charizard.height}}</span>
                </div>
                <div class="information__group">
                    <span class="information__group__title">weight</span>
                    <span class="information__group__description" id="number">{{charizard.weight}}</span>
                </div>

            </div>
        </article>
    </main>
    <footer class="footer">
        <h3 class="footer__title">Evolutions</h3>
        <article class="footer__group">
            <div *ngIf="charmander" class="footer__group__individual hvr-pulse-grow">
                <img [src]="charmander?.sprites?.other?.home?.front_default" [alt]="charmander.name | titlecase"
                    class="footer__group__individual__image" data-id="001">
                <span class="information__group__title" id="number">{{charmander.name | titlecase}}</span>

            </div>
            <div *ngIf="charmeleon" class="footer__group__individual hvr-pulse-grow">
                <img [src]="charmeleon?.sprites?.other?.home?.front_default" [alt]="charmeleon.name | titlecase"
                    class="footer__group__individual__image" data-id="001">
                <span class="information__group__title" id="number">{{charmeleon.name | titlecase}}</span>
            </div>
            <div *ngIf="charizard" class="footer__group__individual hvr-pulse-grow">
                <img [src]="charizard?.sprites?.other?.home?.front_default" [alt]="charizard.name | titlecase"
                    class="footer__group__individual__image" data-id="003">
                <span class="information__group__title" id="number">{{charizard.name | titlecase}}</span>

            </div>
            <!-- <div class="footer__group__individual hvr-pulse-grow">
                <img [src]="charizard?.sprites?.other?.home?.front_default" alt="Bulbasaur"
                    class="footer__group__individual__image" data-id="004">
            </div> -->
        </article>
    </footer>
</div>