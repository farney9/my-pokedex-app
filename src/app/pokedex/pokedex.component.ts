import { Component, OnInit } from '@angular/core';
import { Pokemon } from '../basic/models';
import { PokemonService } from '../basic/services/pokemon.service';

@Component({
  selector: 'app-pokedex',
  templateUrl: './pokedex.component.html',
  styleUrls: ['./pokedex.component.css']
})
export class PokedexComponent implements OnInit {

  charizard?: Pokemon;
  charmander?: Pokemon;
  charmeleon?: Pokemon;

  constructor( private readonly pokemonService:PokemonService) { }

  ngOnInit(): void {
      this.getCharmander(4);
      this.getCharmeleon(5);
      this.getCharizard(6);
  }


  getCharmeleon(id: number) {
    this.pokemonService.getPokemon(id)
    .subscribe(resp => {
      this.charmeleon = resp
      console.log(this.charmeleon.name);
    })
  }

  getCharmander(id: number) {
    this.pokemonService.getPokemon(id)
    .subscribe(resp => {
      this.charmander = resp
      console.log(this.charmander);
    })
  }

  getCharizard(id: number) {
    this.pokemonService.getPokemon(id)
    .subscribe(resp => {
      this.charizard = resp
      console.log(this.charizard.name);
    })
  }
}
