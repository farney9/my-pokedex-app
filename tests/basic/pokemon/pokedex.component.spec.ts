import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { PokemonService } from '../../../src/app/basic/services/pokemon.service';
import { PokedexComponent } from '../../../src/app/pokedex/pokedex.component';

describe('PokedexComponent', () => {
  let component: PokedexComponent;
  let fixture: ComponentFixture<PokedexComponent>;
  let compiled: HTMLElement;
  let service: PokemonService
  let httpMock: HttpTestingController;


  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ PokedexComponent ],
      imports: [HttpClientTestingModule],
      providers: [PokemonService]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(PokedexComponent);
    component = fixture.componentInstance;
    service = TestBed.inject(PokemonService);
    httpMock = TestBed.inject(HttpTestingController)
    fixture.detectChanges();
    compiled = fixture.nativeElement;

  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe hacer match con el snapshot', () => {
    console.log(compiled.innerHTML);
    
    expect(compiled).toMatchSnapshot();
  });


  test('Debe mostrar Loading al cargar la vista', () => {
    const h2 = compiled.querySelector('h2');
    expect(h2?.textContent).toContain('Loading ...')

  });

  /*
  test('Debe cargar a Charizard inmediatamente', () => {
    const pokemonMock = {
      name: 'charizardo!!',
      sprites: {
        other:{
          home: {
            front_default: 'https://charizard.com/front_default.png'
          }
        }
      }
    }

    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/6');
    // esta prueba se asegura que en el ngOnInit se dispare la petición http del servicio pokemon.service para traer la inforamción del pokemon 6 
    // que es charizard

    expect(request.request.method).toBe('GET');

    request.flush(pokemonMock);

    fixture.detectChanges();

    // console.log(compiled.innerHTML);

    const h3  = compiled.querySelector('h3');
    const img = compiled.querySelector('img');

    expect(h3?.textContent?.toLocaleLowerCase()).toContain(pokemonMock.name.toLocaleLowerCase());
    expect(img?.src).toBe(pokemonMock.sprites.other.home.front_default);
    expect(img?.alt).toBe(pokemonMock.name);

  });
*/
});
