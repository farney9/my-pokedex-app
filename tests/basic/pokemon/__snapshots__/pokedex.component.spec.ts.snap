// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokedexComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <img
        alt="logo"
        class="header__logo"
        src="../../assets/pokeball.png"
      />
      <h1
        class="header__title"
      >
        POKÃ‰DEX
      </h1>
    </header>
    <main
      class="main"
    >
      <div>
        <h2>
          Loading ...
        </h2>
      </div>
      
      <aside
        class="description"
      >
        
      </aside>
      
    </main>
    <footer
      class="footer"
    >
      <h3
        class="footer__title"
      >
        Evolutions
      </h3>
      <article
        class="footer__group"
      >
        
        
        
      </article>
    </footer>
  </div>
</div>
`;
